app/controllers/album_recipients_controller.rb

```diff
@@ -9,8 +9,18 @@ def create
     params[:album_recipient][:email].each do |email|
       @contact = Contact.find_or_create_by!(email: email) # email find if not find from contact then its create first
       album_recipients << @album.album_recipients.create!(album_recipient_params.merge(contact_id: @contact.id).permit!)
+
+      if @album.is_private == false
+        AlbumRecipientMailer.share_public_album_to_recipient_mail( @contact, album_recipients, @album).deliver
```

Comment:
`@album` is same for all recipient, so you can move this `if - else` condition to outer of loop.
Or you can create `service` for this to avoid fat controller.